Formatting fix + new section on how to search for the imported data

Endalaner opened the pull request:
- Updated recommendations for files with multiple feature columns to improve search efficiency.
- Clarified rules for recognizing and organizing measurements per Sample (Library or Preparation) with detailed examples.
- Added new sections on searching for imported data via API, including endpoints for imported data and attached files.
- Included notes on permissions and examples for listing attached files and searching for imported expression data.

Commit:
- Updated recommendations for files with multiple feature columns to improve search efficiency.
- Clarified rules for recognizing and organizing measurements per Sample (Library or Preparation) with detailed examples.
- Added new sections on searching for imported data via API, including endpoints for imported data and attached files.
- Included notes on permissions and examples for listing attached files and searching for imported expression data.

MikhailAf commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
````diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
+
+The response will include all *imported expression data* associated with the study:
+
+```default
````
"```json"

MikhailAf commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
 - [Test_RM.preparations.tsv](https://bio-test-data.s3.amazonaws.com/Research_Model_BR-205/Test_RM.preparations.tsv), a TSV file with information about sample preparations
```
Could you please place Libraries and Preparations after Samples, it would be nicer to have them there

MikhailAf commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
+
+ODM allows you to add data, which conceptually falls into two categories:
+
+- **Imported data**
```
Indexed data (Expression, Variants, Flow Cytometry)

MikhailAf commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
+
+### Attached Files Endpoints
+
+Attached Files
```
No need in this line

MikhailAf commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
+### Attached Files Endpoints
+
+Attached Files
+These are files attached to a study.
```
any non-indexed attachment like PDF, PPT etc.

MikhailAf commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
+- `GET /api/v1/as-user/integration/link/files/by/study/{id}`
+
+!!! note "Importing vs. Attaching"
+    You can attach a valid expression matrix to a study. However, it will not be initialized. "Attachment" is a completely different workflow, and such files cannot be accessed using expression endpoints since they are not indexed.
```
Attachment **has** a completely...

MikhailAf commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
+- **Imported data**
+
+- **Attached files**
+
```
Suggest yours to proceed with this link to review the concept of attachments and current limitations mentioned here user-guide/supported-data/supported-data/#attached-files

eeliane commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
+
+
+!!! note "Permissions"
+    To search for attached files, the study must be public or shared with you. If there are no attached files, the response will be an empty array: [].
```
We don't have currently public studies, so let's use only this part - "must be shared with you"

eeliane commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
+
+
+### Example: Searching for Imported Expression Data
+To find imported (indexed) expression data linked to a particular study, use:
```
Let's mention here endpoint in swagger? I'm a bit confused in general that sometimes we use curls as examples and sometimes we mention swagger endpoints

Endalaner replied:
Should we decide on a single approach or maybe list situation where curl should be presented and where we do not require it?

eeliane replied:
@MikhailAf wdyt? maybe we need to show both endpoint and curl information?

MikhailAf replied:
I suggest to mention as much info as possible, let's mention both

Endalaner replied:
Done, please check

eeliane commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -272,3 +294,92 @@ Libraries metadata is a TSV file with information about how samples were prepare
+    To search for attached files, the study must be public or shared with you. If there are no attached files, the response will be an empty array: [].
+
+
+### Example: Searching for Imported Expression Data
```
I suggest to move this section before information about attachments, because it's a bit confusing that we tell about indexed data, after that we tell about attached data and mention that it's possible to attach expression data, and after that we give an example of indexed expression data search

eeliane reviewed (changes requested):
Good updates :) I left several comments, could you please take a look

Commit:
Fixes for review

Commit:
Update documentation for listing attached files in user guide

- Clarified the endpoint for listing attached files for a study.
- Added an example response in JSON format to illustrate the expected output.
- Included a note on permissions related to accessing attached files.

eeliane commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -20,6 +20,24 @@ Samples metadata is supplied in TSV format file. There needs to be a **Sample So
+
+Libraries metadata is a TSV file with information about how samples were prepared. It contains data related to the quality of samples, barcodes and library properties (single-end vs pair-end). Each sample can have more than 1 corresponding library. Multiple samples can be pooled into the same library, e.g. pooling female and male samples to remove gender-specific signals in the sequencing output  ...
+
+| Sample Source ID   |   Library ID | Library barcode   | Library pool   |
```
Could we please use different IDs for samples and libraries?

Endalaner replied:
Agree, default ones were looking too simple.

eeliane commented on docs/user-guide/doc-odm-user-guide/supported-formats.md:
```diff
@@ -255,20 +295,99 @@ A cross-reference mapping file can be imported. This is a TSV file consisting of
+
+### Attached Files Endpoints
+
+These are any non-indexed attachment like PDF, PPT etc.
```
I'm a bit confused, we mention in the header that this section is about endpoints but we mention description of attached files. Can we maybe move description related to index data and attached files to the section above where we mention these two types?

Endalaner replied:
The original idea was to demonstrate how to query different type of files and provide a short, but clear description on what is the main difference between them. If we are to remove this section, then the section that describes the imported data endpoints should also be cut to keep the consistency:

> ### Imported Data Endpoints
> These are files imported via the GUI or API and linked with entities like samples, libraries etc. Once successfully imported and indexed, they are marked with a checkmark and labeled as “indexed data” in GUI.
> 
> ✅ This data can be searched and queried using run’s, values, metadata, or parent group metadata. For each specific type of data you will need to use different endpoints, such as:

WDYT?

eeliane replied:
Discussed on the call

Commit:
- Added a new example for Libraries metadata in TSV format.
- Clarified the description of attached files to specify non-indexed formats.
- Improved formatting for better readability and consistency.
